src/framework/state/gamestate.ts

```diff
@@ -0,0 +1,92 @@
+import { GameState, PlayerID, PlayerState, Spec } from "../types";
+import { getSpecStarters } from "./spec";
+import { MAX_HAND_SIZE } from "../constants";
+import { shuffleCards } from "../actions/hand";
+
+export type PlayerSetupData = {
+  mainSpec: Spec;
```

Comment:
This is actually incorrect. `mainSpec` is meant to hold the spec choice made at tech 2; it wouldn't be determined from the beginning of the game. Within `GameState`, `mainSpec` is nullable because that choice may not have been made yet.

However, we _do_ need to know the color of starting deck that each player is using. We could specify that either as one of their specs or by passing in the color itself (which may be clearer). The latter would of course require us to verify that the specified color matches at least one of their three specs (although it may be argued that the responsibility for verifying this falls to the calling context, and if the caller wishes to start a game with decks that wouldn't normally be valid then they should be able to).

Reply:
I was going to rely on convention such that the first spec specified in `specs` is the starting deck color, but I saw mainSpec and misunderstood its meaning. I will correct that and go back to my conventions instead.

Reply:
Fixed.